src/libsystemd-network/icmp6-util.c

```diff
@@ -83,6 +85,13 @@ int icmp6_bind_router_solicitation(int index) {
         if (r < 0)
                 return -errno;
 
+        if (!if_indextoname(index, ifname))
```

Comment:
nitpick: unless you actually have a bool or ptr value, please compare numerically, instead of relying on the degrade-to-bool feature of C. See CODING_STYLE. Or in other words, write something like `if (if_indextoname(index, ifname) == 0)…` or so...